File: Programmers/짝지어 제거하기.cpp
#include <string>
#include <stack>
#include <string>
using namespace std;

int solution(string s)
{
    stack<char> st;
    for(int i = 0 ; i < s.size(); i++)
    {
        if(!st.empty()&&st.top() == s[i])
            st.pop();
        else
            st.push(s[i]);
    }
    if(st.empty())  return 1;
    else return 0;
}

File: Programmers/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        Stack<Character> st = new Stack<>();
        for(int i = 0; i < s.length(); i++)
        {
            if(!st.isEmpty() &&s.charAt(i) == st.peek())
                st.pop();
            else
                st.push(s.charAt(i));
        }
        if(st.isEmpty())
            return 1;
        else
            return 0;
    }
}